File: PooCC01CNetz/Semana03/Python/Ejercicios1/Calculadora.py
class Calculadora:
    def ProdPunto(self, V1, V2):
        res = ((V1.getRefX() * V2.getRefX()) + (V1.getRefY() * V2.getRefY()) + (V1.getRefZ() * V2.getRefZ()))
        return res
    
    def ProdCruz(self, V1, V2):
        vec = [0, 0, 0]
        vec[0] = self.Partes(V1.getRefY(), V1.getRefZ(), V2.getRefY(), V2.getRefZ())
        vec[1] = self.Partes(V1.getRefX(), V1.getRefZ(), V2.getRefY(), V2.getRefZ())
        vec[2] = self.Partes(V1.getRefY(), V1.getRefZ(), V2.getRefY(), V2.getRefZ())
        return vec

    def Partes(self, a, b, c, d):
        parte = a * b - c * d
        return parte

File: PooCC01CNetz/Semana3/Java/Ejercicios1/src/Calculadora.java
public class Calculadora {
    private double res;

    public double ProdPunto(Vector3D V1, Vector3D V2) {
        res = ((V1.getRefX() * V2.getRefX()) + (V1.getRefY() * V2.getRefY()) + (V1.getRefZ() * V2.getRefZ()));
        return res;
    }

    public double[] ProdCruz(Vector3D v1, Vector3D v2){
        double vec[] = new double[3];
        vec[0] = Partes(v1.getRefY(), v1.getRefZ(), v2.getRefY(), v2.getRefZ());
        vec[1] = Partes(v1.getRefX(), v1.getRefZ(), v2.getRefY(), v2.getRefZ());
        vec[2] = Partes(v1.getRefY(), v1.getRefZ(), v2.getRefY(), v2.getRefZ());
        return vec;
    }

    public double Partes(double a, double b, double c, double d){
        double parte = a * b - c * d;
        return parte;
    }
}   
